### PR TITLE
Simplifying cw-storage implementation

### PR DESCRIPTION
**Pull request by Buckram123**
Simplifying KeyDeserialize, PrimaryKey and Prefixer for abstract types

### CODE REVIEW

File: framework/packages/abstract-std/src/objects/entry/dex_asset_pairing.rs
Comment (Buckram123) [inline framework/packages/abstract-std/src/objects/entry/dex_asset_pairing.rs:1]: Not sure how to implement PrimaryKey/Prefixer for DexAssetPairing as <(&AssetEntry,  &AssetEntry, &DexName)> and not get borrowing error

**Review by CyberHoward — APPROVED** [review]
Is this state breaking?

**Comment by Buckram123** [comment]
> Is this state breaking?

No snapshot updates so I assume not

File: framework/packages/abstract-std/src/objects/entry/dex_asset_pairing.rs
Comment (Kayanski) [reply framework/packages/abstract-std/src/objects/entry/dex_asset_pairing.rs:1]: Something like that ?
https://github.com/AbstractSDK/abstract/pull/505/commits/8a3b6d26d295cc6cc5a90706841b39a9f33bb2d5

Comment (Buckram123) [reply framework/packages/abstract-std/src/objects/entry/dex_asset_pairing.rs:1]: > Something like that ? [8a3b6d2](https://github.com/AbstractSDK/abstract/commit/8a3b6d26d295cc6cc5a90706841b39a9f33bb2d5) Prefixer and PrimaryKey are automatically implemented for &T when implemented for T

Oh, nice to know. Thanks

**Pull request merged**